docs/source/developer_guide/metadata.md

```diff
@@ -16,50 +16,68 @@ limitations under the License.
-components hosted in remote locations or that need additional metadata in order to connect
-to these external components. 
+# Metadata Services
+While building enterprise extensions for the Elyra project we identified a requirement to integrate with external runtimes, data sources, and other
+components that require additional metadata in order to connect
+to these external components.
 
 The **Metadata Service** provides a generic service that can be used to store metadata
 that can be easily integrated with Elyra backend and/or frontend components.
 
-![Metadata Services](../images/metadata-components.png)
+## Metadata Service Overview
+The core design of the metadata service is that it be primarily driven by _schema_.  That is, all metadata _instances_ must adhere to a schema.  Metadata schema are organized, both logically and physically, into _schemaspaces_ where a given schemaspace _contains_ a set of schemas and each instance of a schemaspace's schemas are co-located relative to the configured storage mechanism.
+
+Externally, both in its persisted and serialized forms, a metadata instance is represented as a JSON structure.   Within the metadata service, however, this JSON structure is represented as a `Metadata` class instance.  The `Metadata` class can be extended via subclassing to enable custom behaviors that are triggered following load operations and/or before and after create, update and delete operations on a _per schema_ level.  That is, it is the respective schema that identifies the name of the class to use as the internal representation.
+
+As previously mentioned, the metadata service has a pluggable storage mechanism.  By default, metadata instances are stored within the file system.  However, alternative storage mechanisms can be configured for, say, NoSQL DB or traditional, table-oriented, databases.
+
+Schemaspaces and their schemas can be dynamically introduced via entrypoints.  When the system starts, it will identify applicable entrypoints and load their corresponding objects to seed the list of _configured_ schemaspaces and schemas.
+
+### Metadata Components
+The Elyra metadata service is composed of several components, many of which were previously referenced, and each responsible for a different task.  
+![MetadataServiceRelationships](../images/metadata-service-relationships.png)
+
+The following briefly describes each of those components.
 
-### Metadata Services structure using the default file system store
+- **[Schemaspace](https://github.com/karlaspuldaro/elyra/blob/d345dbd415cbdb16c8d46f705bcd1b3b6ed54450/elyra/metadata/schema.py#L199)**: This is the top-most component and represents a _container_ of schemas.  It has `id` (UUID), `name` (symbolic name), `display_name` (descriptive name), and `description` properties, along with a set of schemas.  Instances of a schemaspace's schemas are logically grouped together in whatever storage model is in use.  Schemaspaces are discovered and loaded via entrypoints.
```

Comment:
If you perform a global replace of 
```
https://github.com/karlaspuldaro/elyra/blob/d345dbd415cbdb16c8d46f705bcd1b3b6ed54450/elyra/metadata/
```
with
```
https://github.com/elyra-ai/elyra/blob/master/elyra/metadata/
```

this link and subsequent link that currently point to the fork should be valid